cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -884,6 +898,111 @@ impl CoreEnvironment<ReadOnly> {
         self.replace_with(temp_env)?;
         Ok(store_path)
     }
+
+    /// Should not be called with
+    /// !migration_info.needs_manifest_migration && !migration_info.needs_upgrade
+    pub fn migrate_to_v1(
+        &mut self,
+        flox: &Flox,
+        mut migration_info: MigrationInfo,
+    ) -> Result<PathBuf, CoreEnvironmentError> {
```

Comment:
Suggestion: if there's the possibility that this is called with incorrect arguments, it seems like something that should go in a debug log.